lnst/Controller/RunSummaryFormatters/JsonRunSummaryFormatter.py

```diff
@@ -0,0 +1,91 @@
+        ]
+
+        return json.dumps(
+            measurement_results,
+            indent=4 if self.pretty else None,
+        )
+
+    def _transform_result(self, result: Result) -> Optional[dict]:
+        if isinstance(result, JobResult):
+            if isinstance(result.job.what, str):
+                job_info = {
+                    "type": "shell",
+                    "command": result.job.what,
+                }
+            else:
+                job_info = {
+                    "type": "module",
+                    "repr": repr(result.job.what),
+                }
+            return {
+                "type": "job",
+                "action": "start" if isinstance(result, JobStartResult) else "end",
+                "job": job_info,
+            }
+        elif isinstance(result, DeviceMethodCallResult):
+            return {
+                "type": "device_method_call",
+                "host": result.device.host.hostid,
+                "netns": result.device.netns.name if result.device.netns and result.device.netns.name else "",
+                "dev_id": result.device._id,
+                "method": {
+                    "name": result.method_name,
+                    "args": [repr(arg) for arg in result.args],
+                    "kwargs": [f"{k}={v!r}" for k, v in result.kwargs.items()],
+                },
+            }
+        elif isinstance(result, MeasurementResult):
+            if result.measurement_type == "ping":
+                measurement_data = result.data
+            elif result.measurement_type == "cpu":
+                measurement_data = {
+                    "utilization": result.data["cpu"].average,
+                }
+            elif result.measurement_type == "flow":
+                agg_results = result.data["flow_results"]
+                measurement_data = {
+                    "generator_results": agg_results.generator_results.average,
+                    "generator_cpu_stats": agg_results.generator_cpu_stats.average,
+                    "receiver_results": agg_results.receiver_results.average,
+                    "receiver_cpu_stats": agg_results.receiver_cpu_stats.average,
+                }
+            elif result.measurement_type == "tc":
+                measurement_data = {
+                    "rule_install_rate": result.data["rule_install_rate"].average,
+                }
+            elif result.measurement_type == "linuxperf":
+                # linuxperf measurement just generates files
+                return None
+            else:
+                logging.warning(f"unhandled measurement result type: {result.measurement_type}")
+                return None
+            return {
+                "type": "measurement",
+                "measurement_type": result.measurement_type,
+                "data": measurement_data,
+            }
+        elif isinstance(result, Result) and result.data is None:
```

Comment:
imo for generic `Result` we can at least report the `result` and the `description`.

Reply:
looks ok